src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/GenerateDocumentHandlerTest.java

```diff
@@ -393,4 +405,99 @@ void should_not_allow_null_arguments() {
             .hasMessage("callback must not be null")
             .isExactlyInstanceOf(NullPointerException.class);
     }
+
+    @Test
+    void should_send_decision_and_reasons_and_populate_ftpa_field_ada_appeal() {
+        String expectedFtpaDeadlineAda = now.plusDays(ftpaAppealOutOfTimeWorkingDaysAdaAppeal).toString();
```

Comment:
this is a wrong deadline for FTPA. you  are expecting the deadline to be + some calendar days. Instead it should be + working days.
Either hard code the zonedDateTime, mocking that event happened on a certain day and then you calculate the dueDate. For ex: zonedDate = 10.02.2023 and deadlineAda = 17.02.2023.